js/src/forum/components/PostStream.js

```diff
@@ -209,8 +209,10 @@ export default class PostStream extends Component {
       const visibleBottom = Math.min(height, viewportTop + viewportHeight - top);
       const visiblePost = visibleBottom - visibleTop;
 
-      if (top <= viewportTop) {
-        index = parseFloat($this.data('index')) + visibleTop / height;
+      // We take the index of the first item that passed the previous checks.
+      // It is the item that is first visible in the viewport.
+      if (indexFromViewPort === null) {
+        indexFromViewPort = parseFloat($this.data('index')) + visibleTop / height;
```

Comment:
The NaN errors described in https://discuss.flarum.org/d/23667-dev-diary-beta-14/123 and https://discuss.flarum.org/d/23667-dev-diary-beta-14/132, can be probably removed by checking if visibleTop is defined and setting it to 0 if it's not. No idea why it could be undefined though, haven't found a way to reproduce yet. I had this check in my codebase already for beta.13 though, so I believe it's not something new. 

```js
        indexFromViewPort = parseFloat($this.data('index'));
        // Setting visibleTop to zero in rare cases, where it could be undefined.
        let fraction = (visibleTop || 0) / height;
        indexFromViewPort += fraction;
```

Reply:
I believe the NaN issue is moreso the result of other issues than its own thing. #2366 and #2377 should eliminate most of those issues.